Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (id) nibInstantiate
+{
+  return self;
+}
+
+- (IBAction) doAction: (id)sender
+{
+  [_sender performSegueWithIdentifier: _identifier
+                               sender: _sender];
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  NSStoryboardSeguePerformAction *pa = [[NSStoryboardSeguePerformAction allocWithZone: z] init];
+  [pa setTarget: _target];
+  [pa setSelector: [self selector]];
+  [pa setSender: _sender];
+  [pa setIdentifier: _identifier];
+  return pa;
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSTarget"])
+        {
+          [self setTarget: [coder decodeObjectForKey: @"NSTarget"]];
+        }
+      if ([coder containsValueForKey: @"NSSelector"])
+        {
+          [self setSelector: [coder decodeObjectForKey: @"NSSelector"]];
+        }
+      if ([coder containsValueForKey: @"NSSender"])
+        {
+          [self setSender: [coder decodeObjectForKey: @"NSSender"]];
+        }
+      if ([coder containsValueForKey: @"NSIdentifier"])
+        {
+          [self setIdentifier: [coder decodeObjectForKey: @"NSIdentifier"]];
+        }
+      if ([coder containsValueForKey: @"NSKind"])
+        {
+          [self setKind: [coder decodeObjectForKey: @"NSKind"]];
+        }
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  // this is never encoded directly...
+}
+@end
+
+
+@interface NSControllerPlaceholder : NSObject <NSCoding, NSCopying> // , NSSeguePerforming>
+{
+  NSString *_storyboardName;
+  NSStoryboard *_storyboard;
+}
+
+- (NSString *) storyboardName;
+- (void) setStoryboardName: (NSString *)name;
+- (id) instantiate;
+
+@end
+
+@implementation NSControllerPlaceholder
+
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _storyboardName = nil;
+    }
+  return self;
+}
+
+- (NSString *) storyboardName
+{
+  return _storyboardName;
+}
+
+- (void) setStoryboardName: (NSString *)name
+{
+  ASSIGNCOPY(_storyboardName, name);
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  NSControllerPlaceholder *c = [[NSControllerPlaceholder allocWithZone: z] init];
+  [c setStoryboardName: _storyboardName];
+  return c;
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if ([coder allowsKeyedCoding])
+    {
+      if ([coder containsValueForKey: @"NSStoryboardName"])
+        {
+          [self setStoryboardName: [coder decodeObjectForKey: @"NSStoryboardName"]];
+        }
+    }
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  // this is never encoded directly...
+}
+
+- (id) instantiate
+{
+  _storyboard = [NSStoryboard storyboardWithName: _storyboardName
```

Comment:
Could this become a local variable as it is never used as an instance variable?

Reply:
Done